Lib/test/test_free_threading/test_tokenize.py

```diff
@@ -0,0 +1,57 @@
+import io
+import time
+import unittest
+import tokenize
+from functools import partial
+from threading import Thread
+
+from test.support import threading_helper
+
+
+@threading_helper.requires_working_threading()
+class TestTokenize(unittest.TestCase):
+    def test_tokenizer_iter(self):
+        source = io.StringIO("for _ in a:\n  pass")
+        it = tokenize._tokenize.TokenizerIter(source.readline, extra_tokens=False)
+
+        tokens = []
+        def next_token(it):
+            while True:
+                try:
+                    r = next(it)
+                    tokens.append(tokenize.TokenInfo._make(r))
+                    time.sleep(1)
```

Comment:
Can we make this test faster? It's going to run on every CI job. I think we should aim for 0.1 seconds or less for these sorts of tests. (It's currently ~3 seconds).

Reply:
Changed the amount of time each thread sleeps to 0.03 secs, which means that the whole test now takes about 0.1 seconds. Also verified that the test still fails without the fix and succeeds with it.